Adding ES error cases to createError

rclee91 opened the pull request:
Follow up of #63 

Gave createError its own module so we could add cases for ElasticSearch errors.
Updated the tests to reflect the changes as well. 

To test: `pnpm test`

For this change, the ES error will be created into an `http-error` indirectly first, before being sent into `createError` again as a nested error with additional props. 
Doing so this way, we get to keep the ES error `statusCode`, it won't be overwritten. 
However, I have no way of changing the description name of the final error to whatever ES might have named the specific status code to. The names are based on the `status` code that `createError` receives.
For example, a `404` status code will always be created as a  `NotFoundError`, as shown below
```js
    const elasticError = new errors.ResponseError({
      body: { error: 'testing ElasticSearch Error' },
      statusCode: 404,
    });

    const testESError = createError(503, elasticError);
    console.log(testESError);
```
The log would be something like
```text
 NotFoundError: ElasticSearch Error:ResponseError
        at ... {
      body: { error: 'testing ElasticSearch Error' }
    }
```
So, it might be up for discussion if we want to keep the ES error status codes, or if we have it so they can be overwritten.

humphd commented on src/createError.js:
```diff
@@ -0,0 +1,22 @@
+const _createError = require('http-errors');
+const { errors } = require('@elastic/elasticsearch');
+
+module.exports.createError = (...args) => {
```
`module.exports = (...args) => {`

humphd commented on src/createError.js:
```diff
@@ -0,0 +1,22 @@
+  for (let i = 0; i < args.length; i++) {
+    let arg = args[i];
+    let type = typeof arg;
+    if (type === 'object' && arg instanceof errors.ResponseError) {
```
Add comments please to indicate what each path is doing in this if.

humphd commented on src/index.js:
```diff
@@ -4,7 +4,7 @@ const { createRouter } = require('./app');
 module.exports.logger = require('./logger');
 module.exports.hash = require('./hash');
-module.exports.createError = require('http-errors');
+module.exports.createError = require('./createError').createError;
```
`module.exports.createError = require('./create-error');`

NOTE: always name files in lower-case, since not all filesystems are case sensitive.

humphd commented on src/createError.js:
```diff
@@ -0,0 +1,22 @@
+const _createError = require('http-errors');
```
Rename this file to `src/create-error.js`

rclee91 commented:
Changed the file name, added comments, and changed the `module.exports` line. 
I've also added a new test to show cases where a status code is not included as the first argument.
http-errors will set the status to 500 in such cases.
```js
  test('errors created without a status code should return a status of 500', () => {
    expect(createError('testing').status).toBe(500);
  });
```
